### ISSUE TITLE
irb on CentOS7, JRuby 9.4.2.0 , and JDK 1.8

### ISSUE THREAD

**Issue by hiroyuki-sato** [jruby/jruby#7759]
It is similar #7740. But the error message doesn't match the same. I report just in case.

**Environment Information**
  
Provide at least:
* JRuby version : 9.4.2.0
* Operating system and platform CentOS7 (3.10.0-1160.59.1.el7.x86_64)
* Java 1.8

**Expected Behavior**

`java -jar ./jruby-complete-9.4.2.0.jar -S irb`

Prompt `irb(main):001:0>`

**Actual Behavior**

A Prompt does not show correctly.

```
▽^[[2;2R
```

And Press `^C` and Return, I got the following error.

```
▽^[[4;2RErrno::EBADF: Bad file descriptor - channel: org.jruby.util.io.ChannelFD@31b82e0f
                          getc at org/jruby/RubyIO.java:2945
                    cursor_pos at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/reline/ansi.rb:252
                          loop at org/jruby/RubyKernel.java:1586
                    cursor_pos at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/reline/ansi.rb:251
                           raw at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/io/console/ffi/stub_console.rb:6
                    cursor_pos at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/reline/ansi.rb:248
  may_req_ambiguous_char_width at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/reline.rb:478
                inner_readline at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/reline.rb:291
                 readmultiline at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/reline.rb:254
                 readmultiline at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/forwardable.rb:238
                 readmultiline at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/forwardable.rb:238
                          gets at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/input-method.rb:421
                    eval_input at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:491
                 signal_status at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:722
                    eval_input at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:490
                           lex at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/ruby-lex.rb:273
      each_top_level_statement at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/ruby-lex.rb:242
                          loop at org/jruby/RubyKernel.java:1586
      each_top_level_statement at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/ruby-lex.rb:239
                         catch at org/jruby/RubyKernel.java:1292
      each_top_level_statement at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/ruby-lex.rb:238
                    eval_input at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:509
                           run at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:443
                         catch at org/jruby/RubyKernel.java:1292
                           run at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:442
                         start at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:375
                        <main> at uri:classloader:/META-INF/jruby.home/bin/jirb:13
```

**Comment by enebo** [comment]
@hiroyuki-sato for 9.4 we switched to using reline gem for readline/cursor stuff.   It is a major issue that ffi cannot load.  The error message is bad :)  I opened https://github.com/jruby/jruby/issues/7760 for that.

I think this may be the same problem as https://github.com/jruby/jruby/issues/7579

**Comment by headius** [comment]
A workaround for you: run IRB with `--readline` flag for now. We are actively working on fixing console issues on AArch64.

**Comment by hiroyuki-sato** [comment]
Hello, @headius. Thank you for your comment.

It seems to work well(without `--readline` flag) JRuby `jruby-complete-9.4.5.0.jar` on CentOS 7(x86_64) (original issue environment)
Thanks

```
java -jar jruby-complete-9.4.5.0.jar -S irb
irb(main):001:0>
irb(main):002:0>
irb(main):003:0>
```

**Comment by headius** [comment]
We marked #7579 as fixed in 9.4.3.0 but based on responses in jnr/jffi#138 it was never fully fixed. We also have still-open #7695 which was intended to handle additional platforms.

There's some new leads in jnr/jffi#138, however, and we might be on to a proper fix very soon (9.4.6.0 timeframe).

**Comment by headius** [comment]
Another clarification...

> A Prompt does not show correctly.

This is similar to issues with IRB on Apple M1 due to the FFI-based io-console not working properly there. It is just a symptom of the underlying FFI logic not loading properly on CentOS (which we should also handle better, but at least we can fix the CentOS issue in jnr/jffi#138).

**Comment by headius** [comment]
The core issue here is fixed by the JNR update in #8105. The remaining discussion is about the failure of IRB when native support fails, which can be worked around by forcing IRB back to readline or a simpler UI without reline.
